sumo-api/sumo/metrics/tests/mesh_evaluator_tests.py

```diff
@@ -0,0 +1,104 @@
+"""
+Copyright (c) Facebook, Inc. and its affiliates.
+
+This source code is licensed under the MIT license found in the
+LICENSE file in the root directory of this source tree.
+
+Test MeshEvaluator functions
+
+TODO:
+More tests:
+Explicitly test sample_mesh / sample_element.  Currently, this
+is implicitly tested in the constructor. 
+
+Explicitly test data association.  Add test to verify that
+removing an element generates a false positive and detecting an
+extra element generates a missed detection.  Add test to
+check that moving an element gives lowered geometric scores
+
+Visualize sampled point cloud and colors to verify that they are
+correct.
+"""
+from easydict import EasyDict as edict
+import numpy as np
+import os
+import unittest
+
+from sumo.geometry.pose3 import Pose3
+from sumo.semantic.project_scene import ProjectScene
+from sumo.metrics.evaluator import Evaluator
+from sumo.metrics.mesh_evaluator import MeshEvaluator
+
+class TestMeshEvaluator(unittest.TestCase):
+    def setUp(self):
+        """
+        Common setup for test cases
+        """
+        self.data_path = os.path.join(os.getcwd(), 'sumo/metrics/test_data')
+        self.ground_truth = ProjectScene.load(self.data_path, 'meshes_sample')
+        self.submission = ProjectScene.load(self.data_path, 'meshes_sample')
+        self.settings = Evaluator.default_settings()
+        self.settings.categories = ['wall', 'floor', 'ceiling', 'sofa', 'coffee_table']
+        self.settings.density = 100
+        
+    def test_shape_similarity(self):
+        """
+        Verify that the shape similarity measure is producing sane outputs.
+        """
+        
+        # make a dummy scene
+        scene = ProjectScene("meshes")
+
+        # TODO: Get deepcopy working for ProjectScene and make a simpler example for faster unit test.
```

Comment:
Is this dependent on GltfModel deep copy and has it been addressed?

Reply:
We have a deepcopy, but it does not work like we need it to to be able to make copies of ProjectScenes